### PR TITLE
access: fix diag raised when privileges are granted not from admin

### LINKED ISSUE #5389: Assertion during grant
I have Ubuntu 18.04.
I have Tarantool 2.5, pulled from source today.
I execute these requests:
```
box.cfg{listen=3302}
box.schema.user.grant('guest','read,write,create','universe')
box.session.su('guest')
box.schema.user.create("optimizer")
box.schema.user.grant("optimizer", "alter", "space")
```

Result:
```
tarantool: /home/pgulutzan/tarantool-2.5/src/box/schema_def.c:86: schema_object_name: Assertion `(int) type < (int) schema_object_type_MAX' failed.
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by Korablev77**
Subj.

Closes #5389

### CODE REVIEW

File: src/box/schema_def.c [hunk 1]
```diff
@@ -86,3 +86,11 @@ schema_object_name(enum schema_object_type type)
 	assert((int) type < (int) schema_object_type_MAX);
 	return object_type_strs[type];
 }
+
+const char *
+schema_entity_name(enum schema_object_type type)
+{
+	assert((int) type > (int) schema_object_type_MAX);
+	assert((int) type < (int) schema_entity_type_MAX);
+	return object_type_strs[(type % (schema_object_type_MAX)) + 1];
```
Comment (alyapunov) [inline src/box/schema_def.c:95]: Please make it consistent with `schema_entity_type` function. I's quite misleading that we convert normal object ID to entity ID via switch and back via arithmetic.
BTW I like arithmetic more.
I would make a couple of functions to convert back and forth and use one of them here.

**Review by alyapunov — CHANGES_REQUESTED** [review]

File: src/box/schema_def.c (hunk 1 above)
Comment (Korablev77) [reply src/box/schema_def.c]: Added a couple of helpers in a separate patch. See on the branch

File: src/box/schema_def.c [hunk 2]
```diff
@@ -46,25 +46,28 @@ static const char *object_type_strs[] = {
 	/* [SC_COLLATION]       = */ "collation",
 };
 
+static enum schema_object_type
```
Comment (alyapunov) [inline src/box/schema_def.c:49]: Comments are needed for both helpers.

File: src/box/schema_def.c [hunk 3]
```diff
@@ -46,25 +46,28 @@ static const char *object_type_strs[] = {
...
+	assert(type >= SC_SPACE);
+	assert((int) type < (int) schema_object_type_MAX);
+	return type + schema_object_type_MAX - 1;
+}
+
+static enum schema_object_type
+schema_entity_type_to_object(enum schema_object_type type)
+{
+	assert((int) type > (int) schema_object_type_MAX);
+	assert((int) type < (int) schema_entity_type_MAX);
```
Comment (alyapunov) [inline src/box/schema_def.c:61]: I bet this will not compile (schema_entity_type_MAX is not defined).

**Review by alyapunov — CHANGES_REQUESTED** [review]

**Review by alyapunov — APPROVED** [review]

**Pull request merged**